app/src/main/java/com/github/swent/echo/data/supabase/SupabaseDataSource.kt

```diff
@@ -279,4 +283,27 @@ class SupabaseDataSource(private val supabase: SupabaseClient) : RemoteDataSourc
     override suspend fun deleteUserProfile(userProfile: UserProfile) {
         supabase.from("user_profiles").delete { filter { eq("user_id", userProfile.userId) } }
     }
+
+    override suspend fun getUserProfilePicture(userId: String): File? {
+        val outputFile =
+            File.createTempFile(userId, ".png", null) // the cache directory is used by default
+        supabase.storage
+            .from("user-profile-picture")
+            .downloadAuthenticatedTo("$userId.png", outputFile)
```

Comment:
I fear that generally profile pictures are real images taken by a camera and not graphics with a relatively limited set of colours. Therefore jpg with it's compression would be preferable over png. What do you think; could this be changed?